### ISSUE TITLE
[ENHANCEMENT] Enable instance label management for harvester_virtualmachine resources

### ISSUE THREAD

**Issue by connorkuehl** [harvester/harvester#4816]
**Is your enhancement related to a problem? Please describe.**

Load Balancers require instance labels to form the backend of the loadbalanced service.

If it were possible to manage Load Balancers under OpenTofu/Terraform, it would not be possible to represent the entire infrastructure end-to-end since the [harvester_virtualmachine schema](https://registry.terraform.io/providers/harvester/harvester/latest/docs/resources/virtualmachine#schema) does not have a way for the user to add labels.

**Describe the solution you'd like**

Add instance labels to the harvester_virtualmachine resource and data source.

**Describe alternatives you've considered**

**Additional context**

**Comment by m-ildefons** [comment]
This is a docs issue. You can add labels to a `harvester_virtualmachine` resource by using Terraform's `tags` attribute.
E.g.:

```
resource "harvester_virtualmachine" "test-labels" {
  name = "test-labels"

  tags = {
    foobar = "barfoo"
  }

[...]
}
```

will result in a `VirtualMachine` resource in Harvester with the label:

```
[...]
metadata:
  labels:
    tag.harvesterhci.io/foobar: "barfoo"
[...]
```

Tbh, I think it's fundamentally wrong that the Harvester provider mingles with the user-provided input in this way, but that's how it currently works.
You should be able to use these labels for the selector of the load balancer regardless.

I'll leave this issue open for now and tackle adding an example to the docs.

@bk201 should we perhaps change the way the Terraform provider works here such that it's a bit more intuitive? For instance, add a separate `labels` subresource to the `harvester_virtualmachine` resource that can be used to define labels verbatim?

**Comment by bk201** [comment]
>You can add labels to a harvester_virtualmachine resource by using Terraform's tags attribute.

@m-ildefons Thanks, we can fix the doc first. I did some research and chatted with @FrankYang0529, "tags" doesn't seem to be a reserved attribute of terraform. Or maybe I misunderstand what you mean. The AWS terraform provider does have this attribute.

Using labels sounds great to me, but is there a way to migrate existing deployments using "tags"?

**Comment by m-ildefons** [comment]
You're right, it seems like there is no reason it's called `tags` rather than `labels` besides that the AWS provider also has `tags`.

I think there is an inconsistency here, because the thing that is called `labels` in Harvester UI is called `tags` in the `.tf` syntax. But this naming is by our own choice and we could change it, or we could provide an equivalent `labels` property for the `.tf` files.
Either would improve the user experience by ensuring consistent vocabulary across the entire system.
Another quirk of the Terraform provider here is that the `tags` aren't applied to the VM resource verbatim, but are transformed (see example). That's why I suggested to add a separate `labels` subresource, which doesn't have this transformation.
Doing so would be backwards compatible, would resolve the inconsistent naming and would provide a way to specify labels without the undocumented and unexpected transformation.
I think this example should demonstrate what I'm suggesting:

```
resource "harvester_virtualmachine" "test-labels" {
  name = "test-labels"

  tags = {
    foobartag = "barfoo"
  }
  
  labels = {
    foobarlabel = "blabla"
  }

[...]
}
```
becomes:
```
apiVersion: kubevirt.io/v1
kind: VirtualMachine
[...]
metadata:
  labels:
    tag.harvesterhci.io/foobartag: "barfoo"
    foobarlabel: "blabla"
[...]

```

Notice how the label would be applied verbatim and the tag is applied with a prefix `tag.harvesterhci.io/`

This would also allow us to keep the old `tags` attribute and avoid migration and/or incompatibilities.

**Comment by owainkenwayucl** [comment]
Having [some sort of property] we could set which didn't prepend the "tag.harvesterhci.io/" would definitely help for our use cases internally.

**Comment by Vicente-Cheng** [comment]
moved to the next version

**Comment by irishgordo** [comment]
Thanks for the enhancement @m-ildefons 😄 !
This all looks good on a `0.0.0-dev` build of the `master` branch of the terraform provider, w/ v1.7.0-rc3 single node

I'll go ahead and close this out

<img width="2879" height="1963" alt="Image" src="https://github.com/user-attachments/assets/2e3c0442-4c79-4664-ab36-5eff92d70d37" />
<img width="2760" height="1948" alt="Image" src="https://github.com/user-attachments/assets/101bf23a-40e0-4ea0-8fb6-2d9f97bb99ed" />
<img width="1558" height="1314" alt="Image" src="https://github.com/user-attachments/assets/89ef94b4-3359-4288-b5ce-c7549de162b9" />
<img width="1328" height="1907" alt="Image" src="https://github.com/user-attachments/assets/4503c549-64c7-46ca-9f37-88520c24fd55" />
<img width="1337" height="1906" alt="Image" src="https://github.com/user-attachments/assets/fa3b4c72-d00c-4258-9cae-3f754651930e" />
<img width="1567" height="1315" alt="Image" src="https://github.com/user-attachments/assets/22ea8847-f69f-4f7b-baf7-f6a8ab4145f0" />
<img width="1336" height="1919" alt="Image" src="https://github.com/user-attachments/assets/af187d94-ff6b-4a9b-8d3c-5816a8ae735e" />
<img width="1537" height="1320" alt="Image" src="https://github.com/user-attachments/assets/5232456d-dafa-4afd-b97b-bfeffc8d3285" />
